src/input_parsers/OnnxParser.cpp

```diff
@@ -0,0 +1,1265 @@
+
+        if (d1 != d2)
+        {
+            String errorMessage = Stringf("Mismatch in tensor dimensions found while broadcasting (%d v %d)", d1, d2);
+            throw MarabouError(MarabouError::ONNX_PARSE_ERROR, errorMessage.ascii());
+        }
+
+        output.insertHead(d1);
+    }
+    return output;
+}
+
+/**
+ * @brief Fills in the variables (values 0 and -1) in the new shape template.
+ * See https://github.com/onnx/onnx/blob/main/docs/Operators.md#Reshape for a
+ * description of the variables.
+ *
+ * @param oldShape the previous shape of the tensor
+ * @param newShapeTemplate the template for the new shape of the tensor
+ * @param allowZero whether or not zero-valued dimensions are allowed. If they
+ * are not then all zeroes are replaced with the corresponding value in the
+ * old shape.
+ * @return
+ */
+TensorShape instantiateReshapeTemplate(TensorShape oldShape, TensorShape newShapeTemplate, bool allowZero)
+{
+    TensorShape newShape;
+    int inferredIndex = -1;
+
+    for (uint i = 0; i < newShapeTemplate.size(); i++)
+    {
+        int dimSize = newShapeTemplate[i];
+        if (dimSize == -1)
+        {
+            // Then this dimension should be inferred.
+            // Temporarily set the dimSize to be 1 so that we
+            // can use the tensorSize function to compute the
+            // size so far.
+            inferredIndex = i;
+            dimSize = 1;
+        }
+        else if (dimSize == 0)
+        {
+            if (!allowZero)
+            {
+                if (i < oldShape.size())
+                {
+                    dimSize = oldShape[i];
+                }
+            }
+        }
+        newShape.append(dimSize);
+    }
+
+    if (inferredIndex != -1)
+    {
+        newShape[inferredIndex] = tensorSize(oldShape) / tensorSize(newShape);
+    }
+    return newShape;
+}
+
+void checkEndianness()
+{
+    int num = 1;
+    bool systemIsLittleEndian = *(char *)&num == 1;
+    if (! systemIsLittleEndian)
+    {
+        String errorMessage = "Support for Onnx files on non-little endian systems is not currently implemented on the command line";
+        throw MarabouError(MarabouError::ONNX_PARSE_ERROR, errorMessage.ascii());
+    }
+}
+
+const onnx::AttributeProto* findAttribute(onnx::NodeProto& node, String name, onnx::AttributeProto_AttributeType expectedType)
+{
+    for (const onnx::AttributeProto& attr : node.attribute())
+    {
+        if (attr.name() == name.ascii())
+        {
+            if (attr.type() != expectedType)
+            {
+                illTypedAttributeError(node, attr, expectedType);
+            }
+            return &attr;
+        }
+    }
+    return nullptr;
+}
+
+float getFloatAttribute(onnx::NodeProto& node, String name, float defaultValue)
+{
+    const onnx::AttributeProto* attr = findAttribute(node, name, onnx::AttributeProto_AttributeType_FLOAT);
+    if (attr == nullptr)
+    {
+        return defaultValue;
+    }
+    return (*attr).f();
```

Comment:
The syntax shorthand we prefer for this expression is attr->f();

Reply:
Also a number of instances below fit this pattern.

Reply:
Fixed everywhere I could find